src/presign/participant.rs

```diff
@@ -1037,7 +1037,8 @@ impl PresignKeyShareAndInfo {
             rng,
         )?;
         let mut transcript = Transcript::new(b"PiLogProof");
-        let secret = ProverSecret::new(&sender_r1_priv.gamma, &sender_r1_priv.nu);
+        let nonce_clone = &sender_r1_priv.nu.get_nonce().clone();
```

Comment:
Here you are taking a `SecretNonce`, accessing the underlying `Nonce` and cloning it. So now we have a copy of `nu` that won't get zeroized. This seems like a security issue? We should either:
1) Fix this to not create a non-Secret copy. The best way to fix this, is probably to have `ProverSecret` use a `SecretNonce` instead of a `Nonce`.
2) If we really do want to convert a `SecretNonce` -> `Nonce` we should have a comment justifying why it is okay.